src/funnels/automation/StableDepositor.sol

```diff
@@ -0,0 +1,210 @@
+        (address gem0, address gem1) = gemA < gemB ? (gemA, gemB) : (gemB, gemA);
+        configs[gem0][gem1] = cfg;
+        emit Config(gemA, gemB, cfg);
+    }
+
+    // Note: the keeper's minAmts value must be updated whenever configs[gem0][gem1] is changed.
+    // Failing to do so may result in this call reverting or in taking on more slippage than intended (up to a limit controlled by configs[src][dst].reqOut).
+    function deposit(address gem0, address gem1, uint128 amt0Min, uint128 amt1Min)
+        toll
+        external
+        returns (uint128 liquidity, uint256 amt0, uint256 amt1)
+    {
+        // Note: we rely on Depositor to enforce gem0 < gem1
+        PairConfig memory cfg = configs[gem0][gem1];
+
+        require(cfg.count > 0, "StableDepositor/exceeds-count");
+        configs[gem0][gem1].count = cfg.count - 1;
+
+        if (amt0Min == 0) amt0Min = cfg.amt0Req;
+        if (amt1Min == 0) amt1Min = cfg.amt1Req;
+        require(amt0Min >= cfg.amt0Req, "StableDepositor/min-amt0-too-small");
+        require(amt1Min >= cfg.amt1Req, "StableDepositor/min-amt1-too-small");
+
+        DepositorLike.LiquidityParams memory p = DepositorLike.LiquidityParams({
+            gem0       : gem0,
+            gem1       : gem1,
+            fee        : cfg.fee,
+            tickLower  : cfg.tickLower,
+            tickUpper  : cfg.tickUpper,
+            liquidity  : 0,             // use desired amount
+            amt0Desired: cfg.amt0,
+            amt1Desired: cfg.amt1,
+            amt0Min    : amt0Min,
+            amt1Min    : amt1Min
+        });
+        (liquidity, amt0, amt1) = depositor.deposit(p);
+    }
+
+    // Note: the keeper's minAmts value must be updated whenever configs[gem0][gem1] is changed.
+    // Failing to do so may result in this call reverting or in taking on more slippage than intended (up to a limit controlled by configs[src][dst].reqOut).
+    function withdraw(address gem0, address gem1, uint128 amt0Min, uint128 amt1Min)
+        toll
+        external
+        returns (uint128 liquidity, uint256 amt0, uint256 amt1)
+    {
+        // Note: we rely on Depositor to enforce gem0 < gem1
+        PairConfig memory cfg = configs[gem0][gem1];
+
+        require(cfg.count > 0, "StableDepositor/exceeds-count");
+        configs[gem0][gem1].count = cfg.count - 1;
+
+        if (amt0Min == 0) amt0Min = cfg.amt0Req;
+        if (amt1Min == 0) amt1Min = cfg.amt1Req;
+        require(amt0Min >= cfg.amt0Req, "StableDepositor/min-amt0-too-small");
+        require(amt1Min >= cfg.amt1Req, "StableDepositor/min-amt1-too-small");
+
+        DepositorLike.LiquidityParams memory p = DepositorLike.LiquidityParams({
+            gem0       : gem0,
+            gem1       : gem1,
+            fee        : cfg.fee,
+            tickLower  : cfg.tickLower,
+            tickUpper  : cfg.tickUpper,
+            liquidity  : 0,             // use desired amount
```

Comment:
I'm wondering if it may be better to calculate the `liquidity` that corresponds to `(cfg.amt0, cfg.amt1)` here and do something like `liquidity = min(pool.positions(key).liquidity, liquidity)` to allow withdrawing the last batch without issue.

Reply:
The problem with reducing the amount (as was in the stable swapper) is that it affects the minOut enforcement, as `minOut` and `reqOut` are not scaled. In theory those can be scaled linearly but you'd have to also call `getAmountsForLiquidity` in order to do that scaling.

I was thinking the facilitator would just call Depositor.withdraw with the remaining liquidity to fully clear the position.

But maybe it would help if we also add `liquidity` to StableDepositor.PairConfig and do the same mutual exclusivity as we did in the Depositor level. Then the facilitator would not have to withdraw directly but instead just set the config for that last withdraw (something it should know how to do anyhow) and allow the keeper to do the call. I guess it depends if we think the facilitator can do the withdraw action himself.

Reply:
Probably fine to let the facilitator manually clear the remaining liquidity. I guess it would be valuable to add some view function giving the amount of liquidity in the position (as well as amt0, amt1 in the position, for convenience), especially since we don't use the NFT any more so we can't rely on existing UIs.

Reply:
Yeah I couldn't find a UI for that, everything is using the nft position manager.
As discussed we will explore in a separate PR.